src/contrib/cluster/Akka.Cluster.Sharding/ShardRegion.cs

```diff
@@ -553,16 +560,28 @@ private void ReplyToRegionStatsQuery(IActorRef sender)
             return Task.WhenAll(tasks);
         }
 
+        private List<ActorSelection> GracefulShutdownCoordinatorSelections
```

Comment:
Over thinking it here IMHO. The property allocates a new list because the `MembersByAge` value will be determined at the time this method is called, and it only gets called once during shutdown so allocations / performance isn't an issue. All it's really doing is hiding the implementation details for how the next best coordinators are likely to be picked.